### PR TITLE
Replace ID/PDA/Wallet checks with GetID()

### PR DESCRIPTION
**Pull request by JohnWildkins**
as title

the tl;dr is that there should be far fewer places where using something like a wristbound computer or a wallet isn't recognized as a proper ID holder

~~and there's definitely no reason at all why I specifically targeted anything looking for /obj/item/device/pda, no sir~~

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Comment by JohnWildkins** [comment]
!bugfix

**Comment by JohnWildkins** [comment]
also yes I did test (most) of these changes, the only ones going untested are things like the revolutionary demote check thing which I couldn't be bothered to set up the nine different things required to see if that works

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -639,12 +635,9 @@
...
 
-			if(wear_id)
-				if(istype(wear_id,/obj/item/card/id))
-					perpname = wear_id:registered_name
-				else if(istype(wear_id,/obj/item/device/pda))
-					var/obj/item/device/pda/tempPda = wear_id
-					perpname = tempPda.owner
+			if(GetIdCard())
+				var/obj/item/card/id/id = GetIdCard()
+				perpname = id:registered_name
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:640]:
```suggestion
				perpname = id.registered_name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -664,12 +657,9 @@
...
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:662]:
```suggestion
				perpname = id.registered_name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -689,12 +679,9 @@
...
 			var/perpname = "wot"
-			if(wear_id)
-				if(istype(wear_id,/obj/item/card/id))
-					perpname = wear_id:registered_name
-				else if(istype(wear_id,/obj/item/device/pda))
-					var/obj/item/device/pda/tempPda = wear_id
-					perpname = tempPda.owner
+			if(GetIdCard())
+				var/obj/item/card/id/id = GetIdCard()
+				perpname = id:registered_name
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:684]:
```suggestion
				perpname = id.registered_name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -714,12 +701,9 @@
...
 
-			if(wear_id)
-				if(istype(wear_id,/obj/item/card/id))
-					perpname = wear_id:registered_name
-				else if(istype(wear_id,/obj/item/device/pda))
-					var/obj/item/device/pda/tempPda = wear_id
-					perpname = tempPda.owner
+			if(GetIdCard())
+				var/obj/item/card/id/id = GetIdCard()
+				perpname = id:registered_name
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:706]:
```suggestion
				perpname = id.registered_name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -749,12 +733,9 @@
...
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:738]:
```suggestion
				perpname = id.registered_name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -775,12 +756,9 @@
...
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:761]:
```suggestion
				perpname = id.registered_name
```
why the fuck did whoever wrote this not make this a proc

File: code/modules/mob/living/carbon/human/human.dm [hunk 7]
```diff
@@ -800,12 +778,9 @@
...
 			var/perpname = "wot"
-			if(wear_id)
-				if(istype(wear_id,/obj/item/card/id))
-					perpname = wear_id:registered_name
-				else if(istype(wear_id,/obj/item/device/pda))
-					var/obj/item/device/pda/tempPda = wear_id
-					perpname = tempPda.owner
+			if(GetIdCard())
+				var/obj/item/card/id/id = GetIdCard()
+				perpname = id:registered_name
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human.dm:783]:
```suggestion
				perpname = id.registered_name
```

**Review by skull132 — APPROVED** [review]

**Pull request merged**